### PR TITLE
Added custom IAM role for API Gateway logs

### LINKED ISSUE #6459: Option to specify api gateway log IAM role instead of creating one
# This is a Bug Report

## Description

- What went wrong?
getting this cloudformation error:
API:   iam:CreateRole User:   arn:aws:sts::xxxx:assumed-role/xxxx/AWSCodeBuild-xxx-xxx-xxx-xxx-xxx   is not authorized to perform: iam:CreateRole on resource:   arn:aws:iam::xxx:role/yyy-dev-us-east-1-apiGatewayLogsRole   with an explicit deny
- What did you expect should have happened?
there should be an option to specify the role to use for api gateway logging
- What was the config you used?
  logs:
    restApi: true
- What stacktrace or error message from your provider did you see?
see above

Similar or dependent issues:

- #6134 : it's closed but don't see the resolution

## Additional Data

- **_Serverless Framework Version you're using_**:
- **_Operating System_**:
- **_Stack Trace_**:
- **_Provider Error messages_**:

### PR DESCRIPTION
**Pull request by miguel-a-calles-mba**
## What did you implement:

Closes #6459

Creates a custom IAM for API Gateway logging since all API Gateways share one global IAM role. 

## How did you implement it:

Added a new field to the `serverless.yml` file that allows specifying a custom role.

```yaml
provider:
  logs:
    restApi: true # Optional configuration which specifies if API Gateway logs are used
    restApiIamRole: custom-iam-role-name # Optional configuration which specifies if API Gateway logs IAM role name to create and use
```

## How can we verify it:

1. Create two Serverless stacks
2. Add the following to each `serverless.yml` file:
```yaml
provider:
  logs:
    restApi: true
    restApiIamRole: custom-iam-role-name
```
3. Deploy both stacks in different stages, but to the same AWS account
4. Go to the [API Gateway settings](https://console.aws.amazon.com/apigateway/home?region=us-east-1#/settings)
5. Confirm the value `arn:aws:iam::514209761729:role/custom-iam-role-name` exists for the CloudWatch log role ARN.

## Todos:

_**Note: Run `npm run test-ci` to run all validation checks on proposed changes**_

- [ ] Write tests and confirm existing functionality is not broken.  
       **Validate via `npm test`**
- [ ] Write documentation
- [ ] Ensure there are no lint errors.  
       **Validate via `npm run lint-updated`**  
       _Note: Some reported issues can be automatically fixed by running `npm run lint:fix`_
- [ ] Ensure introduced changes match Prettier formatting.  
       **Validate via `npm run prettier-check-updated`**  
       _Note: All reported issues can be automatically fixed by running `npm run prettify-updated`_
- [ ] Make sure code coverage hasn't dropped
- [ ] Provide verification config / commands / resources
- [ ] Enable "Allow edits from maintainers" for this PR
- [ ] Update the messages below

**_Is this ready for review?:_** NO  
**_Is it a breaking change?:_** NO

### CODE REVIEW

File: lib/plugins/aws/package/compile/events/apiGateway/lib/stage.js [hunk 1]
```diff
@@ -114,6 +117,14 @@ function getLogGroupResource(service, stage) {
 }
 
 function getIamRoleResource(service, stage) {
```
Comment (pmuens) [inline lib/plugins/aws/package/compile/events/apiGateway/lib/stage.js]: We should rename `service` here so that it's clear that this method can handle bot:

1. Referencing the correct role which is auto-created
1. Re-using the role which could be passed-in

**Comment by miguel-a-calles-mba** [comment]
> I just looked through the code and added some comments. How come the role is just the name and not an `arn`? Is there a need to create this role previously? Isn't this change also creating an `AWS::IAM::Role` resource?

Good idea to accept an ARN as well. I will incorporate that.

> I guess my confusion is that I don't understand the underlying issue. Can you rephrase the problem?

Let's say you deploy a service `myfirstservice` in `us-east-1` region, the deployment creates an IAM role `myfirstservice-us-east-1-apiGatewayLogsRole` and sets the global API Gateway logging role to that.

Let's say you deploy another service 'mysecondservice' in the same region, the deployment creates an IAM role `mysecondservice-us-east-1-apiGatewayLogsRole` and replaces the global API Gateway logging role to that. Now both `myfirstservice` and `mysecondservice` stacks are using `mysecondservice-us-east-1-apiGatewayLogsRole` because it's global to the account and region.

This PR allows someone to specify the global IAM role name to avoid creating an "X" number of IAM  roles for an "X" number of stacks.

File: lib/plugins/aws/package/compile/events/apiGateway/lib/stage.js (hunk 1 above)
Comment (miguel-a-calles-mba) [reply lib/plugins/aws/package/compile/events/apiGateway/lib/stage.js]: I committed a change, which hopefully addresses your comments.

**Comment by miguel-a-calles-mba** [comment]
Hi @pmuens, makes sense. I addressed your comments.

**Comment by medikoo** [comment]
@miguel-a-calles-mba with https://github.com/serverless/serverless/pull/6531 we've improved current state of things, so that it's the only one IAM role created for that (no matter how many different stacks you deploy to different regions)

Still I believe a possibility to use an existing role (that was created aside by user) is still a feature worth having, so this PR is still very welcome. Unfortunately #6531 introduced a conflicts for this PR, if you need any help with this let us now.

**Comment by miguel-a-calles-mba** [comment]
@medikoo, sounds good. I'll see what I can do. Thanks.
